In [350]:
import pandas as pd

In [351]:
adverse = pd.read_csv('adverse_reaction.csv')
treatment = pd.read_csv('treatment.csv')
patient = pd.read_csv('patients.csv')

In [352]:
adverse_cpy = adverse.copy()
treatment_cpy = treatment.copy()
patient_cpy = patient.copy()

In [353]:
print(adverse.shape)
print(treatment.shape)
print(patient.shape)

(34, 3)
(280, 7)
(503, 14)


- Here, PATIENT is the csv files of details of all patients who are giving human test for insulin usage.
- In TREATMENT file, the auralin means who used insulin dose via tablet and novadra means who used through syringe.
    41u - 48u means lower unit is 41 and upper unit is 48 for that patient.
- ADVERSE file has all the patients who had adverse effect while using it. 

### we have chosen 350 patients (175 auralin + 175 novadra) 

In [354]:
patient

patient_id assigned_sex given_name      surname  \
0             1       female        Zoe      Wellish   
1             2       female     Pamela         Hill   
2             3         male        Jae       Debord   
3             4         male       Liêm         Phan   
4             5         male        Tim      Neudorf   
..          ...          ...        ...          ...   
498         499         male    Mustafa    Lindström   
499         500         male      Ruman      Bisliev   
500         501       female      Jinke    de Keizer   
501         502       female    Chidalu  Onyekaozulu   
502         503         male        Pat      Gersten   

                       address               city       state  zip_code  \
0         576 Brown Bear Drive  Rancho California  California   92390.0   
1    2370 University Hill Road          Armstrong    Illinois   61812.0   
2        1493 Poling Farm Road               York    Nebraska   68467.0   
3          2335 Webster Street         Woodbridge          NJ    7095.0   
4         1428 Turkey Pen Lane             Dothan          AL   36303.0   
..                         ...                ...         ...       ...   
498        2530 Victoria Court       Milton Mills          ME    3852.0   
499   494 Clarksburg Park Road             Sedona          AZ   86341.0   
500          649 Nutter Street      Overland Park          MO   64110.0   
501   3652 Boone Crockett Lane            Seattle          WA   98109.0   
502          2778 North Avenue               Burr    Nebraska   68324.0   

           country                                          contact  \
0    United States             951-719-9170ZoeWellish@superrito.com   
1    United States            PamelaSHill@cuvox.de+1 (217) 569-3204   
2    United States                 402-363-6804JaeMDebord@gustr.com   
3    United States       PhanBaLiem@jourrapide.com+1 (732) 636-8246   
4    United States                  334-515-7487TimNeudorf@cuvox.de   
..             ...                                              ...   
498  United States      207-477-0579MustafaLindstrom@jourrapide.com   
499  United States               928-284-4492RumanBisliev@gustr.com   
500  United States            816-223-6007JinkedeKeizer@teleworm.us   
501  United States  ChidaluOnyekaozulu@jourrapide.com1 360 443 2060   
502  United States             PatrickGersten@rhyta.com402-848-4923   

     birthdate  weight  height   bmi  
0    7/10/1976   121.7      66  19.6  
1     4/3/1967   118.8      66  19.2  
2    2/19/1980   177.8      71  24.8  
3    7/26/1951   220.9      70  31.7  
4    2/18/1928   192.3      27  26.1  
..         ...     ...     ...   ...  
498  4/10/1959   181.1      72  24.6  
499  3/26/1948   239.6      70  34.4  
500  1/13/1971   171.2      67  26.8  
501  2/13/1952   176.9      67  27.7  
502   5/3/1954   138.2      71  19.3  

[503 rows x 14 columns]

In [355]:
treatment

given_name       surname    auralin    novodra  hba1c_start  hba1c_end  \
0     veronika      jindrová  41u - 48u          -         7.63       7.20   
1       elliot    richardson          -  40u - 45u         7.56       7.09   
2     yukitaka      takenaka          -  39u - 36u         7.68       7.25   
3         skye    gormanston  33u - 36u          -         7.97       7.62   
4       alissa        montez          -  33u - 29u         7.78       7.46   
..         ...           ...        ...        ...          ...        ...   
275     albina      zetticci  45u - 51u          -         7.93       7.73   
276       john   teichelmann          -  49u - 49u         7.90       7.58   
277     mathea       lillebø  23u - 36u          -         9.04       8.67   
278     vallie        prince  31u - 38u          -         7.64       7.28   
279     samúel  guðbrandsson  53u - 56u          -         8.00       7.64   

     hba1c_change  
0             NaN  
1            0.97  
2             NaN  
3            0.35  
4            0.32  
..            ...  
275          0.20  
276           NaN  
277          0.37  
278          0.36  
279          0.36  

[280 rows x 7 columns]

In [356]:
adverse

given_name     surname           adverse_reaction
0         berta  napolitani  injection site discomfort
1          lena        baer               hypoglycemia
2        joseph         day               hypoglycemia
3        flavia  fiorentino                      cough
4       manouck     wubbels          throat irritation
5       jasmine       sykes               hypoglycemia
6        louise     johnson               hypoglycemia
7       albinca     komavec               hypoglycemia
8           noe      aranda               hypoglycemia
9         sofia   hermansen  injection site discomfort
10        tegan     johnson                   headache
11         abel     yonatan                      cough
12    abdul-nur         isa               hypoglycemia
13         leon      scholz  injection site discomfort
14     gabriele     saenger               hypoglycemia
15       jia li        teng                     nausea
16        jakob    jakobsen               hypoglycemia
17  christopher    woodward                     nausea
18          ole    petersen               hypoglycemia
19       finley    chandler                   headache
20     anenechi       chidi               hypoglycemia
21     miłosław  wiśniewski  injection site discomfort
22        lixue       hsueh  injection site discomfort
23        merci      leroux               hypoglycemia
24         kang         mai  injection site discomfort
25       elliot  richardson               hypoglycemia
26      clinton      miller          throat irritation
27       idalia       moore               hypoglycemia
28       xiuxiu       chang               hypoglycemia
29         alex    crawford               hypoglycemia
30       monika      lončar               hypoglycemia
31       steven         roy                   headache
32      cecilie      nilsen               hypoglycemia
33    krisztina      magyar               hypoglycemia

In [357]:
# Now let's do it programmatically
patient.info()
# We should see columnName and its Dtype, and change if it doesnot satisfies. 
# Object type means String 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 503 entries, 0 to 502
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   patient_id    503 non-null    int64  
 1   assigned_sex  503 non-null    object 
 2   given_name    503 non-null    object 
 3   surname       503 non-null    object 
 4   address       491 non-null    object 
 5   city          491 non-null    object 
 6   state         491 non-null    object 
 7   zip_code      491 non-null    float64
 8   country       491 non-null    object 
 9   contact       491 non-null    object 
 10  birthdate     503 non-null    object 
 11  weight        503 non-null    float64
 12  height        503 non-null    int64  
 13  bmi           503 non-null    float64
dtypes: float64(3), int64(2), object(9)
memory usage: 55.1+ KB


In [358]:
#Since we have 6 columns which have missing values. Let's see that values
patient[patient['address'].isnull()]

patient_id assigned_sex given_name      surname address city state  \
209         210       female     Lalita  Eldarkhanov     NaN  NaN   NaN   
219         220         male         Mỹ        Quynh     NaN  NaN   NaN   
230         231       female  Elisabeth      Knudsen     NaN  NaN   NaN   
234         235       female    Martina    Tománková     NaN  NaN   NaN   
242         243         male       John      O'Brian     NaN  NaN   NaN   
249         250         male   Benjamin       Mehler     NaN  NaN   NaN   
257         258         male        Jin         Kung     NaN  NaN   NaN   
264         265       female   Wafiyyah       Asfour     NaN  NaN   NaN   
269         270       female     Flavia   Fiorentino     NaN  NaN   NaN   
278         279       female   Generosa        Cabán     NaN  NaN   NaN   
286         287         male      Lewis         Webb     NaN  NaN   NaN   
296         297       female        Chỉ          Lâm     NaN  NaN   NaN   

     zip_code country contact   birthdate  weight  height   bmi  
209       NaN     NaN     NaN   8/14/1950   143.4      62  26.2  
219       NaN     NaN     NaN    4/9/1978   237.8      69  35.1  
230       NaN     NaN     NaN   9/23/1976   165.9      63  29.4  
234       NaN     NaN     NaN    4/7/1936   199.5      65  33.2  
242       NaN     NaN     NaN   2/25/1957   205.3      74  26.4  
249       NaN     NaN     NaN  10/30/1951   146.5      69  21.6  
257       NaN     NaN     NaN   5/17/1995   231.7      69  34.2  
264       NaN     NaN     NaN   11/3/1989   158.6      63  28.1  
269       NaN     NaN     NaN   10/9/1937   175.2      61  33.1  
278       NaN     NaN     NaN  12/16/1962   124.3      69  18.4  
286       NaN     NaN     NaN    4/1/1979   155.3      68  23.6  
296       NaN     NaN     NaN   5/14/1990   181.1      63  32.1

In [359]:
patient.describe()
# here the weight min weight is 48 pound and height is 27 inches, ie error, which can create wrong calculations 

patient_id      zip_code      weight      height         bmi
count  503.000000    491.000000  503.000000  503.000000  503.000000
mean   252.000000  49084.118126  173.434990   66.634195   27.483897
std    145.347859  30265.807442   33.916741    4.411297    5.276438
min      1.000000   1002.000000   48.800000   27.000000   17.100000
25%    126.500000  21920.500000  149.300000   63.000000   23.300000
50%    252.000000  48057.000000  175.300000   67.000000   27.200000
75%    377.500000  75679.000000  199.500000   70.000000   31.750000
max    503.000000  99701.000000  255.900000   79.000000   37.700000

In [360]:
# checking duplicate data
patient[patient['address'].duplicated()]

patient_id assigned_sex given_name     surname             address  \
29           30         male       Jake    Jakobsen   648 Old Dear Lane   
219         220         male         Mỹ       Quynh                 NaN   
229         230         male       John         Doe     123 Main Street   
230         231       female  Elisabeth     Knudsen                 NaN   
234         235       female    Martina   Tománková                 NaN   
237         238         male       John         Doe     123 Main Street   
242         243         male       John     O'Brian                 NaN   
244         245         male       John         Doe     123 Main Street   
249         250         male   Benjamin      Mehler                 NaN   
251         252         male       John         Doe     123 Main Street   
257         258         male        Jin        Kung                 NaN   
264         265       female   Wafiyyah      Asfour                 NaN   
269         270       female     Flavia  Fiorentino                 NaN   
277         278         male       John         Doe     123 Main Street   
278         279       female   Generosa       Cabán                 NaN   
282         283       female      Sandy      Taylor  2476 Fulton Street   
286         287         male      Lewis        Webb                 NaN   
296         297       female        Chỉ         Lâm                 NaN   
502         503         male        Pat     Gersten   2778 North Avenue   

            city     state  zip_code        country  \
29   Port Jervis  New York   12771.0  United States   
219          NaN       NaN       NaN            NaN   
229     New York        NY   12345.0  United States   
230          NaN       NaN       NaN            NaN   
234          NaN       NaN       NaN            NaN   
237     New York        NY   12345.0  United States   
242          NaN       NaN       NaN            NaN   
244     New York        NY   12345.0  United States   
249          NaN       NaN       NaN            NaN   
251     New York        NY   12345.0  United States   
257          NaN       NaN       NaN            NaN   
264          NaN       NaN       NaN            NaN   
269          NaN       NaN       NaN            NaN   
277     New York        NY   12345.0  United States   
278          NaN       NaN       NaN            NaN   
282     Rainelle        WV   25962.0  United States   
286          NaN       NaN       NaN            NaN   
296          NaN       NaN       NaN            NaN   
502         Burr  Nebraska   68324.0  United States   

                                        contact   birthdate  weight  height  \
29   JakobCJakobsen@einrot.com+1 (845) 858-7707    8/1/1985   155.8      67   
219                                         NaN    4/9/1978   237.8      69   
229                 johndoe@email.com1234567890    1/1/1975   180.0      72   
230                                         NaN   9/23/1976   165.9      63   
234                                         NaN    4/7/1936   199.5      65   
237                 johndoe@email.com1234567890    1/1/1975   180.0      72   
242                                         NaN   2/25/1957   205.3      74   
244                 johndoe@email.com1234567890    1/1/1975   180.0      72   
249                                         NaN  10/30/1951   146.5      69   
251                 johndoe@email.com1234567890    1/1/1975   180.0      72   
257                                         NaN   5/17/1995   231.7      69   
264                                         NaN   11/3/1989   158.6      63   
269                                         NaN   10/9/1937   175.2      61   
277                 johndoe@email.com1234567890    1/1/1975   180.0      72   
278                                         NaN  12/16/1962   124.3      69   
282        304-438-2648SandraCTaylor@dayrep.com  10/23/1960   206.1      64   
286                                         NaN    4/1/1979 

In [361]:
treatment.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 280 entries, 0 to 279
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   given_name    280 non-null    object 
 1   surname       280 non-null    object 
 2   auralin       280 non-null    object 
 3   novodra       280 non-null    object 
 4   hba1c_start   280 non-null    float64
 5   hba1c_end     280 non-null    float64
 6   hba1c_change  171 non-null    float64
dtypes: float64(3), object(4)
memory usage: 15.4+ KB


In [362]:
treatment.describe()

hba1c_start   hba1c_end  hba1c_change
count   280.000000  280.000000    171.000000
mean      7.985929    7.589286      0.546023
std       0.568638    0.569672      0.279555
min       7.500000    7.010000      0.200000
25%       7.660000    7.270000      0.340000
50%       7.800000    7.420000      0.380000
75%       7.970000    7.570000      0.920000
max       9.950000    9.580000      0.990000

In [363]:
adverse.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34 entries, 0 to 33
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   given_name        34 non-null     object
 1   surname           34 non-null     object
 2   adverse_reaction  34 non-null     object
dtypes: object(3)
memory usage: 944.0+ bytes


# Quality(Dirty Data) issues:

[Completeness(cm), Validity(V), Accuracy(A), consistance(icn)]

### `patient`

- some state have full name whereas some state have short forms (C)
- some zipcode is four digit where as some are five (V)
- zip code are stored in float type (V)
- we have NaN values from different columns (cm)
- Sex should be changed to categorical Dtype and birthdate should be changed Date type 
- Error data for weight and height (A)
- inconsistant way of storing phone numbers (ICN)
- Duplicated and not useful data (ICN)

### `treatment`

- hbac1c col has missing/incorrect value(CM)
- novodra and auralin adding 41u - 48u has made auraln column a string (V)
- only 280 rows are present(A)
- in patient file, first letter of name are capital but in treatment all names are small (CM) 
- incorrect datatype for auralin and novodra datatypes (V) 
- usage of dashes instead of NaN in auralin and Novadra (ICN)

### `adverse`

# Tidiness Issues:

### `patient`

- contact col has both email and number

### `treatment`

- 3 variable stored in auralin | novodra, lower dose and upper dose 

### `adverse`

- no point of using this table

# Cleaning Process

you should follow the following process:

- Completeness Issues
- Tideness Issues
- Remaining Data quality issues like validity, accuracy, and consistancy

steps involved in data cleaning

- Define
- Code
- Test

## Completeness Issues
1. hba1c change DONE
2. capitalize first letter of name in treatment file DONE

In [364]:
treatment_cpy['hba1c_change'] = treatment_cpy['hba1c_start'] - treatment_cpy['hba1c_end']  
for i in  range(len(treatment_cpy['given_name'])):

    treatment_cpy.loc[i,  'surname'] = treatment_cpy['surname'][i].capitalize()
    treatment_cpy.loc[i,  'given_name'] = treatment_cpy['given_name'][i].capitalize()

treatment_cpy

given_name       surname    auralin    novodra  hba1c_start  hba1c_end  \
0     Veronika      Jindrová  41u - 48u          -         7.63       7.20   
1       Elliot    Richardson          -  40u - 45u         7.56       7.09   
2     Yukitaka      Takenaka          -  39u - 36u         7.68       7.25   
3         Skye    Gormanston  33u - 36u          -         7.97       7.62   
4       Alissa        Montez          -  33u - 29u         7.78       7.46   
..         ...           ...        ...        ...          ...        ...   
275     Albina      Zetticci  45u - 51u          -         7.93       7.73   
276       John   Teichelmann          -  49u - 49u         7.90       7.58   
277     Mathea       Lillebø  23u - 36u          -         9.04       8.67   
278     Vallie        Prince  31u - 38u          -         7.64       7.28   
279     Samúel  Guðbrandsson  53u - 56u          -         8.00       7.64   

     hba1c_change  
0            0.43  
1            0.47  
2            0.43  
3            0.35  
4            0.32  
..            ...  
275          0.20  
276          0.32  
277          0.37  
278          0.36  
279          0.36  

[280 rows x 7 columns]

In [365]:
#next problem we want to solve is we want to have treatment column where auralin and novodra will stored
        #source         #columns to leaved unchanged                                                                   #new col with col name       #col name where values are stored of changed columns 
treatment_cpy  = pd.melt(treatment_cpy, id_vars=['given_name', 'surname', 'hba1c_start', 'hba1c_end', 'hba1c_change'],  var_name = 'treatment', value_name = "dose")
treatment_cpy

given_name       surname  hba1c_start  hba1c_end  hba1c_change treatment  \
0     Veronika      Jindrová         7.63       7.20          0.43   auralin   
1       Elliot    Richardson         7.56       7.09          0.47   auralin   
2     Yukitaka      Takenaka         7.68       7.25          0.43   auralin   
3         Skye    Gormanston         7.97       7.62          0.35   auralin   
4       Alissa        Montez         7.78       7.46          0.32   auralin   
..         ...           ...          ...        ...           ...       ...   
555     Albina      Zetticci         7.93       7.73          0.20   novodra   
556       John   Teichelmann         7.90       7.58          0.32   novodra   
557     Mathea       Lillebø         9.04       8.67          0.37   novodra   
558     Vallie        Prince         7.64       7.28          0.36   novodra   
559     Samúel  Guðbrandsson         8.00       7.64          0.36   novodra   

          dose  
0    41u - 48u  
1            -  
2            -  
3    33u - 36u  
4            -  
..         ...  
555          -  
556  49u - 49u  
557          -  
558          -  
559          -  

[560 rows x 7 columns]

In [366]:
treatment_cpy = treatment_cpy[treatment_cpy["dose"] != "-"] 
treatment_cpy

given_name      surname  hba1c_start  hba1c_end  hba1c_change treatment  \
0     Veronika     Jindrová         7.63       7.20          0.43   auralin   
3         Skye   Gormanston         7.97       7.62          0.35   auralin   
6       Sophia       Haugen         7.65       7.27          0.38   auralin   
7        Eddie       Archer         7.89       7.55          0.34   auralin   
9         Asia      Woźniak         7.76       7.37          0.39   auralin   
..         ...          ...          ...        ...           ...       ...   
548     Yasmin       Araujo         7.82       7.36          0.46   novodra   
549     Hiromu     Horikawa         7.77       7.28          0.49   novodra   
551        Leo       Vieira         7.74       7.36          0.38   novodra   
552     Steven          Roy         7.87       7.43          0.44   novodra   
556       John  Teichelmann         7.90       7.58          0.32   novodra   

          dose  
0    41u - 48u  
3    33u - 36u  
6    37u - 42u  
7    31u - 38u  
9    30u - 36u  
..         ...  
548  51u - 54u  
549  47u - 46u  
551  30u - 33u  
552  41u - 43u  
556  49u - 49u  

[280 rows x 7 columns]

In [367]:
#since dose is object we convert into string using "str" function
#here expand = True in split returns a DataFrame
# if you get all the warning just use the following code
treatment_cpy._is_copy = False
treatment_cpy[['start','end']] = treatment_cpy['dose'].str.split('-', expand = True)
treatment_cpy

given_name      surname  hba1c_start  hba1c_end  hba1c_change treatment  \
0     Veronika     Jindrová         7.63       7.20          0.43   auralin   
3         Skye   Gormanston         7.97       7.62          0.35   auralin   
6       Sophia       Haugen         7.65       7.27          0.38   auralin   
7        Eddie       Archer         7.89       7.55          0.34   auralin   
9         Asia      Woźniak         7.76       7.37          0.39   auralin   
..         ...          ...          ...        ...           ...       ...   
548     Yasmin       Araujo         7.82       7.36          0.46   novodra   
549     Hiromu     Horikawa         7.77       7.28          0.49   novodra   
551        Leo       Vieira         7.74       7.36          0.38   novodra   
552     Steven          Roy         7.87       7.43          0.44   novodra   
556       John  Teichelmann         7.90       7.58          0.32   novodra   

          dose start   end  
0    41u - 48u  41u    48u  
3    33u - 36u  33u    36u  
6    37u - 42u  37u    42u  
7    31u - 38u  31u    38u  
9    30u - 36u  30u    36u  
..         ...   ...   ...  
548  51u - 54u  51u    54u  
549  47u - 46u  47u    46u  
551  30u - 33u  30u    33u  
552  41u - 43u  41u    43u  
556  49u - 49u  49u    49u  

[280 rows x 9 columns]

In [368]:
treatment_cpy.drop(columns = [ 'dose'], inplace=True)
treatment_cpy = treatment_cpy.drop_duplicates()
treatment_cpy

given_name      surname  hba1c_start  hba1c_end  hba1c_change treatment  \
0     Veronika     Jindrová         7.63       7.20          0.43   auralin   
3         Skye   Gormanston         7.97       7.62          0.35   auralin   
6       Sophia       Haugen         7.65       7.27          0.38   auralin   
7        Eddie       Archer         7.89       7.55          0.34   auralin   
9         Asia      Woźniak         7.76       7.37          0.39   auralin   
..         ...          ...          ...        ...           ...       ...   
548     Yasmin       Araujo         7.82       7.36          0.46   novodra   
549     Hiromu     Horikawa         7.77       7.28          0.49   novodra   
551        Leo       Vieira         7.74       7.36          0.38   novodra   
552     Steven          Roy         7.87       7.43          0.44   novodra   
556       John  Teichelmann         7.90       7.58          0.32   novodra   

    start   end  
0    41u    48u  
3    33u    36u  
6    37u    42u  
7    31u    38u  
9    30u    36u  
..    ...   ...  
548  51u    54u  
549  47u    46u  
551  30u    33u  
552  41u    43u  
556  49u    49u  

[279 rows x 8 columns]

In [369]:
def isNaN(string):
    return string != string
patient["contact"][14]

'AsiaWozniak@rhyta.com918-712-3469'

In [370]:
import re
rgx = r'\d'
pattern = re.compile(rgx)
patternS = re.compile(r'[^\d\-\(\)\+]')
contact = []
email = []
for i in range(len(patient)):
    string = patient['contact'][i]
    if(isNaN(string)): 
        contact.append("No Contact")
        email.append("No Email")
        continue
    matchI = pattern.findall(string)
    matchS = patternS.findall(string)
    em = "".join(matchS)
    n = len(matchI)
    con = ''
#     print(matchI)
    if n == 11 :
        con = "(+" + matchI[0] + ") "
        for k in range(1, n):
            con+=matchI[k]
            if (k == 3 or k == 6 ):
                con+="-"
    elif n==10:
        for k in range(n):
            con += matchI[k]
            if (k == 2 or k == 5 ):
                con += "-"
    contact.append(con)
    email.append(em)

In [371]:
patient['email'] = email
patient['number'] = contact
patient.drop(columns = ["contact"], inplace = True)
patient

patient_id assigned_sex given_name      surname  \
0             1       female        Zoe      Wellish   
1             2       female     Pamela         Hill   
2             3         male        Jae       Debord   
3             4         male       Liêm         Phan   
4             5         male        Tim      Neudorf   
..          ...          ...        ...          ...   
498         499         male    Mustafa    Lindström   
499         500         male      Ruman      Bisliev   
500         501       female      Jinke    de Keizer   
501         502       female    Chidalu  Onyekaozulu   
502         503         male        Pat      Gersten   

                       address               city       state  zip_code  \
0         576 Brown Bear Drive  Rancho California  California   92390.0   
1    2370 University Hill Road          Armstrong    Illinois   61812.0   
2        1493 Poling Farm Road               York    Nebraska   68467.0   
3          2335 Webster Street         Woodbridge          NJ    7095.0   
4         1428 Turkey Pen Lane             Dothan          AL   36303.0   
..                         ...                ...         ...       ...   
498        2530 Victoria Court       Milton Mills          ME    3852.0   
499   494 Clarksburg Park Road             Sedona          AZ   86341.0   
500          649 Nutter Street      Overland Park          MO   64110.0   
501   3652 Boone Crockett Lane            Seattle          WA   98109.0   
502          2778 North Avenue               Burr    Nebraska   68324.0   

           country  birthdate  weight  height   bmi  \
0    United States  7/10/1976   121.7      66  19.6   
1    United States   4/3/1967   118.8      66  19.2   
2    United States  2/19/1980   177.8      71  24.8   
3    United States  7/26/1951   220.9      70  31.7   
4    United States  2/18/1928   192.3      27  26.1   
..             ...        ...     ...     ...   ...   
498  United States  4/10/1959   181.1      72  24.6   
499  United States  3/26/1948   239.6      70  34.4   
500  United States  1/13/1971   171.2      67  26.8   
501  United States  2/13/1952   176.9      67  27.7   
502  United States   5/3/1954   138.2      71  19.3   

                                    email             number  
0                ZoeWellish@superrito.com       951-719-9170  
1                  PamelaSHill@cuvox.de    (+1) 217-569-3204  
2                    JaeMDebord@gustr.com       402-363-6804  
3             PhanBaLiem@jourrapide.com    (+1) 732-636-8246  
4                     TimNeudorf@cuvox.de       334-515-7487  
..                                    ...                ...  
498       MustafaLindstrom@jourrapide.com       207-477-0579  
499                RumanBisliev@gustr.com       928-284-4492  
500             JinkedeKeizer@teleworm.us       816-223-6007  
501  ChidaluOnyekaozulu@jourrapide.com     (+1) 360-443-2060  
502              PatrickGersten@rhyta.com       402-848-4923  

[503 rows x 15 columns]